hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreFlusher.java

```diff
@@ -127,20 +127,22 @@ public MemStoreFlusher(final Configuration conf,
     this.blockingWaitTime = conf.getInt("hbase.hstore.blockingWaitTime",
       90000);
     int handlerCount = conf.getInt("hbase.hstore.flusher.count", 2);
-    if (handlerCount < 1) {
-      LOG.warn("hbase.hstore.flusher.count was configed to {} which is less than 1, corrected to 1",
-          handlerCount);
-      handlerCount = 1;
+    if (server != null) {
+      if (handlerCount < 1) {
+        LOG.warn("hbase.hstore.flusher.count was configed to {} which is less than 1, "
+            + "corrected to 1", handlerCount);
+        handlerCount = 1;
+      }
+      LOG.info("globalMemStoreLimit="
+          + TraditionalBinaryPrefix
+              .long2String(this.server.getRegionServerAccounting().getGlobalMemStoreLimit(), "", 1)
+          + ", globalMemStoreLimitLowMark="
+          + TraditionalBinaryPrefix.long2String(
+            this.server.getRegionServerAccounting().getGlobalMemStoreLimitLowMark(), "", 1)
+          + ", Offheap="
+          + (this.server.getRegionServerAccounting().isOffheap()));
     }
     this.flushHandlers = new FlushHandler[handlerCount];
```

Comment:
If I read this right, handlerCount could be < 1 here if server == null?

Reply:
Yeah, in TestMemStoreFlusher, we don't want the handler to run, so i set the hbase.hstore.flusher.count to 0 in conf.

Reply:
If making a new PR, do you want to move the handlerCount check for < 1 outside of the if server != null paren? Seems like a generally applicable good test... no need to keep it exclusive to when the server is non-null?

Reply:
If the server is null, the handler could not work actually, so we do not need to create them?